modules/ROOT/pages/batch-reference.adoc

```diff
@@ -0,0 +1,241 @@
++
+* `NO_FAILURES`: Default setting. Accepts a record for processing if the record never failed to process successfully within any preceding Batch Step component or if there is no upstream Batch Step component.
+* ONLY_FAILURES:  Accepts a record for processing if the record failed to process successfully within any preceding Batch Step component.
+* ALL: Accepts a record for processing regardless of processing successfully or failing within any preceding Batch Step component.
+
+|===
+
+////
+  // TODO: The MetaData tab, same as for other batch components.
+  // POINT TO DOC ON THIS. CREATE THE DOC IF IT DOESN'T EXIST YET
+  // OR IS INCOMPLETE. Would need work item for this task.
+////
+
+[[ref_batch_aggregator_component]]
+== Batch Aggregator Component (<batch:aggregator)
+
+The Batch Aggregator component is an optional component that acts on an array of records. Aggregation takes place during the xref:batch-processing-concept.adoc#phase_process[Process phase] of a batch job instance.
+
+There is no default configuration for the Batch Aggregator component. If you use this component, you must indicate whether to stream records (`streaming`) or pull records from the queue into separate arrays of a fixed size (`size`), but not both. The settings are mutually exclusive. _Streamed records are non-repeatable_, which means that random access to streamed records is not possible.
+
+The Batch Aggregator component accepts a single processor for the message `payload`. A downstream sibling to that processor receives no `payload` but can gain access to the value of Mule variables, such as the ID of the batch job instance, using `vars.batchJobInstanceId`. In addition, a scope within the aggregator, such as xref:for-each-scope-concept.adoc[For Each] scope, that iterates over each record in the array can contain multiple child processors.
+
+The Batch Aggregator component also provides a way preserve the MIME type of records (`preserveMimeTypes`), which is needed when accessing part of a record that you are aggregating as a fixed size (`size`). For details about record processing within this component, see batch-processing-concept.adoc#batch_aggregator_processing[batch aggregator processing].
+
+Note that some connectors, such as xref:salesforce-connector::salesforce-connector-reference.adoc#upsert[Anypoint Connector for Salesforce (Salesforce Connector)] and xref:netsuite-connector::netsuite-reference.adoc#Upsert[Anypoint Connector for NetSuite (NetSuite Connector)], provide operations that can handle record-level errors without causing the batch aggregation process to fail.
```

Comment:
upsert is one

others TODO

Reply:
Current docs refer only to upsert (from https://docs.mulesoft.com/mule-runtime/4.4/batch-filters-and-batch-aggregator#batch-aggregator-characteristics):

> > Several Anypoint Connectors can handle record-level errors without failing a whole batch aggregation (i.e., upsert).
> > At runtime, these connectors keep track of which records were successfully accepted by the target resource, and which failed to upsert. Thus, rather than failing a complete group of records, the connector upserts as many records as it can, and tracks any failures for notification. Some of these connectors are:
> Salesforce
> 
> NetSuite
> 
> To make sure that the connector you are using supports record-level errors, check the connector’s documentation

Reply:
See comment in edit mode.

Reply:
TODO: create new story to add specific operations

Reply:
done, also lists the other TODOs [W-11934285]